app/models/apple/podcast_container.rb

```diff
@@ -222,8 +257,27 @@ def missing_podcast_audio?
     end
 
     def needs_delivery?
-      # TODO: Overwriting the podcast audio with another file
-      missing_podcast_audio? && podcast_deliveries.empty?
+      # Handle the case where the podcast container *does* have podcast audio,
+      # but doesn't have any podcast deliveries / files. This is a weird edge
+      # case but it amounts to checking the deliveries to see if any are there.
+      # If there are no deliveries, then the code that polls/checks the delivery
+      # status will fail.
+      podcast_deliveries.empty?
+    end
+
+    def source_url_expires_at
+      return nil if source_url.blank?
+
+      uri = URI.parse(source_url)
+      query_params = CGI.parse(uri.query)
+      timestamp = query_params["exp"].first
+      Time.at(timestamp.to_i).utc
+    end
+
+    def source_url_expired?
```

Comment:
Wait, how does that happen?  Are you holding onto a `dovetail-cdn` url for multiple days?

Generally, we should always be going back to `dovetail.prxu.org` (DTR) to get a fresh redirect, before doing anything with that file.  `dovetail-cdn` urls aren't meant to be long-lived.

Reply:
I refactored the source reset routine so that it runs with each re-upload attempt.  It's a bit aggressive in this way, because the delivery files are the sentinels that determine if another upload should be attempted.  But with this approach there is not really an alternative. Resetting the source CDN url for the upload data means that the delivery file is invalidated. So it is destroyed.